src/Microsoft.Diagnostics.Runtime/src/DataReaders/MacOS/MacOSProcessDataReader.cs

```diff
@@ -221,10 +225,62 @@ private unsafe T Read<T>(ref ulong address)
             return result;
         }
 
-        public bool GetThreadContext(uint threadID, uint contextFlags, Span<byte> context)
+        public unsafe bool GetThreadContext(uint threadID, uint contextFlags, Span<byte> context)
         {
-            // TODO
-            return false;
+            LoadThreads();
+            if (!_threadActs.TryGetValue(threadID, out var threadAct))
+                return false;
+
+            int regSize = sizeof(x86_thread_state64_t);
+
+            byte[] buffer = ArrayPool<byte>.Shared.Rent(regSize);
+            try
+            {
+                fixed (byte* data = buffer)
+                {
+                    int stateCount = Native.x86_THREAD_STATE64_COUNT;
+                    int kr = Native.thread_get_state(threadAct, Native.x86_THREAD_STATE64, new IntPtr(data), ref stateCount);
+                    if (kr != 0)
+                        throw new ClrDiagnosticsException($"thread_get_state failed with status code 0x{kr:x}");
```

Comment:
If we can't get the context, you should return `false` here instead of throwing.